taxi/tests/test_forms.py

```diff
@@ -0,0 +1,141 @@
+            username="Driver1",
+            password="TestPassword123",
+            first_name="Test",
+            last_name="Driver1",
+            license_number="AAA11111"
+        )
+        self.driver2 = get_user_model().objects.create_user(
+            username="Driver2",
+            password="TestPassword123",
+            first_name="Test",
+            last_name="Driver2",
+            license_number="AAA22222"
+        )
+        self.driver3 = get_user_model().objects.create_user(
+            username="Driver3",
+            password="TestPassword123",
+            first_name="Test",
+            last_name="Driver3",
+            license_number="AAA33333"
+        )
+
+    def test_driver_search_by_username(self):
+        response = self.client.get(reverse("taxi:driver-list") + '?username=Driver1')
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Driver1")
+        self.assertNotContains(response, "Driver2")
+        self.assertNotContains(response, "Driver3")
+
+    def test_driver_search_by_partial_username(self):
+        response = self.client.get(reverse("taxi:driver-list") + '?username=Driver2')
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Driver2")
+        self.assertNotContains(response, "Driver1")
+        self.assertNotContains(response, "Driver3")
+
+    def test_driver_search_no_results(self):
+        response = self.client.get(reverse("taxi:driver-list") + '?username=Driver4')
+        self.assertEqual(response.status_code, 200)
+        self.assertNotContains(response, "Driver1")
+        self.assertNotContains(response, "Driver2")
+        self.assertNotContains(response, "Driver3")
+
+
+class CarSearchTest(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="TestUsername",
+            password="TestPassword123"
+        )
+        self.client.force_login(self.user)
+        self.manufacturer1 = Manufacturer.objects.create(name="Toyota", country="Japan")
+        self.manufacturer2 = Manufacturer.objects.create(name="Ford", country="USA")
+        self.manufacturer3 = Manufacturer.objects.create(name="Tesla", country="USA")
+        self.car1 = Car.objects.create(model="Camry", manufacturer=self.manufacturer1)
+        self.car2 = Car.objects.create(model="F150", manufacturer=self.manufacturer2)
+        self.car3 = Car.objects.create(model="Model S", manufacturer=self.manufacturer3)
+
+    def test_car_search_by_model(self):
+        response = self.client.get(reverse("taxi:car-list") + '?model=Camry')
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Camry")
+        self.assertNotContains(response, "F150")
+        self.assertNotContains(response, "Model S")
+
+    def test_car_search_by_partial_model(self):
+        response = self.client.get(reverse("taxi:car-list") + '?model=F150')
```

Comment:
The test name `test_car_search_by_partial_model` suggests that it should test partial matches, but the query `?model=F150` is a full model name. Consider changing the query to a partial model name to match the test name.